lib/src/html_renderer.dart

```diff
@@ -91,7 +120,24 @@ class HtmlRenderer implements NodeVisitor {
   }
 
   void visitElementAfter(Element element) {
+    if (const ['blockquote', 'ul', 'ol'].contains(element.tag)) {
```

Comment:
Haha again with the weird rule. Commonmark citation?

Reply:
It's all about blocks within blocks. Updated the logic. Much better now.